server/src/main/java/org/elasticsearch/snapshots/SnapshotShutdownProgressTracker.java

```diff
@@ -85,6 +87,12 @@ public class SnapshotShutdownProgressTracker {
     private final AtomicLong abortedCount = new AtomicLong();
     private final AtomicLong pausedCount = new AtomicLong();
 
+    /**
+     * Ensure that we only log a periodic progress report while there are in progress snapshots,
+     * and that we do not continue to log after all snapshots have completed
+     */
+    private boolean shouldLogPeriodicProgressReport = false;
```

Comment:
This needs to be volatile. I also don't think this is safe to change before canceling the scheduled logProgressReport(): you could set it to false before cancelling, and race with logProgressReport()'s assert that it is true.

This boolean need only represent completion of exit log message, prevent a double exit message somehow. There are already other safeguards for concurrency in the other methods.